core/src/main/java/io/aiven/kafka/tieredstorage/KafkaRemoteStorageManager.java

```diff
@@ -148,7 +149,7 @@ class KafkaRemoteStorageManager extends InternalRemoteStorageManager {
             .findFirst()
             .orElse(null);
 
-        this.segmentManifestCache = new MemorySegmentManifestCache(config.storage(), mapper);
+        this.segmentManifestCache = new MemorySegmentManifestCache(storage, mapper);
```

Comment:
I don't recall if there is any purpose on having separate storage instances for fetching manifests. It seems ok to me.

CC @AnatolyPopov in case you know of any special case.